imports/lib/schemas/Hunt.ts

```diff
@@ -44,6 +45,24 @@ const HuntFields = t.type({
   memberDiscordRole: t.union([SavedDiscordObjectFields, t.undefined]),
 });
 
+const SavedDiscordObjectPattern = {
+  id: String,
+  name: String,
+};
+
+export const HuntPattern = {
```

Comment:
Yeah I did try to make this work. Basically, you want something like

```ts
function codecToPattern<
  T extends Record<string, any>,
  P extends Record<keyof T, t.Mixed>
>(codec: t.InterfaceType<P, T>): Match.Pattern<T>;
```

and then just walk the codec similar to what `buildSchema` does. I ran into some challenges typing it the first time but I should have another go at it. (I think the first step is to actually export `Match.Pattern`, which is currently private to the type declaration, but that should be fixable.)

Reply:
OK I got sufficiently nerd-sniped to take another stab at this and I think I have something that works: https://tsplay.dev/w2P7Vm (scroll down to the bottom and look at the type of `DocumentCodecPattern` and `x` after the call to `check`).

It should work with everything we've done to date except for `SettingCodec`, because it can't handle a tagged union intersected with another interface type (I'm not sure that's even representable in a single `Match` pattern anyway - you'd need a `Match.AllOf` or something)

I'm not going to complexify this PR by incorporating this but I may come back to it later, and it's there in case you want to play with it.